### ISSUE TITLE
sct_extract_metric : CSF management + output an image for visual check

### ISSUE THREAD

**Issue by tanguyduval** [neuropoly/spinalcordtoolbox#958]
Usually the CSF is not clean in the data because of vertebra/bones. Also in quantitative metrics we  compute the metric in the SC mask only --> no values or absurd values in the CSF.
For instance when I was doing my computation in the different tracts for fh. The value in the CSF found is 0 while it whould be 1.. Which bias all other metrics values...

Two solutions to prevent this error:
1- Output a .png/.nii with a synthetic SC slice generated from results of sct_extract_metrics. Like in this example:
![image](https://cloud.githubusercontent.com/assets/7785316/17749870/97c1c4a8-648d-11e6-976a-45456ab68699.png)
Advise the user to check the CSF value in this output image

2- add an option to fix the value of the CSF. In this case the MAP equation will be splitted in two:
`||y - P\{csf}.X\{csf} - P_csf.x_csf||` with x_csf the value of the CSF..

**Comment by tanguyduval** [comment]
image updated..

**Comment by tanguyduval** [comment]
Concerning the coding the idea is to substract P_csf.x_csf from the image (with P_csf the tract CSF and x_csf the fixed value we vant to attribute). Then run MAP or ML without the tract CSF.

**Comment by tanguyduval** [comment]
Example of failure in the current implementation:
![image](https://cloud.githubusercontent.com/assets/7785316/17780747/788e7060-653a-11e6-849c-055088bfebca.png)
![image](https://cloud.githubusercontent.com/assets/7785316/17780752/7e4d54e4-653a-11e6-82c2-bf273f314857.png)

**Comment by slevyrosetti** [comment]
![demo ml](https://cloud.githubusercontent.com/assets/7769691/17787419/13c67868-6557-11e6-87d7-0e13dd8d927d.jpg)

**Comment by slevyrosetti** [comment]
A flag to fix the value of a specific label has been added (`-fix 31,0` to fix the CSF value to 0 for example) on branch `sl_sct_extract_metric` (commit f081e5e).

**Comment by slevyrosetti** [comment]
I added the flag `-output-map` to generate a metric map based on the estimated metric value within each label and the fractional volumes of each label.
branch `sl_extract_metric` commit `7d9f2a3`.

**Comment by jcohenadad** [comment]
i'm not a big fan of the solution-- i would rather add the possibility to input a mask (e.g. could be mask that only shows the cord, or the GM), and then multiply the matrix X (volume fraction of each labels) by the mask. In that case, better NOT to use MAP or ML.

**Comment by slevyrosetti** [comment]
I agree, this option is more dedicated to advanced user but I think it is very useful, especially to handle cases such as the previously published image, where the cord almost reaches the bone and the CSF space is really thin. If you prefer, we could make this option invisible for the "standard" user, would you prefer that?

Furthermore, note that if we do not use MAP or ML, the misregistration of the CSF does not affect the metric estimation within the WM tracts, the metric estimations in the CSF and in the WM tracts are independent. Thus, this option is already available! :)

**Comment by tanguyduval** [comment]
I agree with simon, a mask would be useless if we don't use MAP or ML.
However your solution Julien is similar to the weighted MAP (issue #1018 ): in this case we also have to multiply y by this weighted mask.
Cons:
- The lateral tracts (spinothalamic) will be highly biased because they have lot of partial volume with CSF.

**Comment by jcohenadad** [comment]
for visual output, use the new qc module from sct_propseg in: https://github.com/3324fr/spinalcordtoolbox/blob/master/scripts/sct_propseg.py

**Comment by slevyrosetti** [comment]
Waiting the end of the QC project.
